### PR TITLE
memory function unit tests

### PR DESCRIPTION
**Pull request by EnderSmith**
TL;DR:
- NEW FILE `diceRollUnitTests.html` (duplicate of `diceRoll/index.html`)
- REMOVED links to `diceRollUnitTests.js` from `diceRoll/index.html`.
- REMOVED references to `diceRollUnitTests.js` from `diceRoll.js`.
- REMOVED out of date unit tests from `diceRollUnitTests.js`.
- ADDED new memory function unit tests to `diceRollUnitTests.js` (6 total).
- RENAMED `savePreloaded()` to `restoreDefaultSaveItems()`.
- ADDED `return` values for `simulateFirstVisit()`, `restoreDefaultSaveItems()`, and `loadMemory()` functions.
- ADDED `testTF` property to `deleteSaveItem()`, to override `confirm()` alert.

HTMLpreview:

http://htmlpreview.github.io/?https://github.com/EnderSmith/EnderSmith.github.io/blob/unitTesting/diceRoll/diceRollUnitTests.html

### CODE REVIEW

File: diceRoll/diceRoll2.js [hunk 1]
```diff
@@ -90,13 +89,14 @@ function addSaveItemListeners() {
...
+
       new SaveItemListeners(saved[saved_props[i]].id, saved[saved_props[i]].rollArray);
     }
 }
 
 // -- testable functions --
 
-// save
+// memory
 function SaveItem(id, name, rollArray) {
```
Comment (ultimatetag) [inline diceRoll/diceRoll2.js:100]: Side question: Why is SavedItem capitalized?

Comment (ultimatetag) [reply diceRoll/diceRoll2.js:100]: (I mean SaveItem)

File: diceRoll/diceRoll2.js [hunk 2]
```diff
@@ -106,19 +106,19 @@ function SaveItem(id, name, rollArray) {
  }
  localStorage.saved = JSON.stringify(copyOfSaved);
  loadMemory();
+ // SaveItemListeners(copyOfSaved[id].id, copyOfSaved[id].rollArray);
```
Comment (ultimatetag) [inline diceRoll/diceRoll2.js:109]: Remove old commented code (you can look at history of you need it later)

File: diceRoll/diceRoll2.js [hunk 3]
```diff
@@ -141,30 +141,36 @@ function loadMemory() {
...
+   return savedMenu;
 }
-function savePreloaded() {
+function restoreDefaultSaveItems() {
  var saved = preloaded;
  localStorage.saved = JSON.stringify(saved);
+ return saved;
 }
-function deleteSaveItem(id) {
+function deleteSaveItem(id, testTF) {
```
Comment (ultimatetag) [inline diceRoll/diceRoll2.js:153]: What does TF stand for?

File: diceRoll/diceRollUnitTests.html [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  <body>
+    <div id="wholepage" class="holder">
+      <div class="row banner"></div>
+      <div id="calcDisp" class="disp col-m-12 col-t-6 col-4">
+        <div id="dispIn"></div>
+        <div id="dispOut"></div>
+        <div id="dispEng"></div>
+      </div>
+      <div id="calcHolder" class="holder col-m-12 col-t-6 col-4">
+        <!-- <div class="row">
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.html:24]: Remove commented code

File: diceRoll/diceRollUnitTests.html [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+          </button>
+          <button id="rollBtn" class="btn col-m-7 col-t-7 col-7">
+            roll
+          </button>
+          <button id="savedBtn" class="btn col-m-3 col-t-3 col-3">
+            saved
+          </button>
+        </div>
+      </div>
+      <!-- <br>
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.html:118]: Remove commented code

File: diceRoll/diceRollUnitTests.js [hunk 6]
```diff
@@ -1,53 +1,20 @@
...
-      new UnitTest('keypadPress minus', keypadPress('-', true), '-0'),
-      new UnitTest('keypadPress number after sign', keypadPress(10, true), '-10'),
-      new UnitTest('keypadPress minus before plus', keypadPress('-', true), '-0'),
-      new UnitTest('keypadPress plus after minus', keypadPress('+', true), '0'),
-      new UnitTest('keypadPress dice before number', keypadPress('d20', true), '1d20'),
-      new UnitTest('keypadPress number after dice', keypadPress(3, true), '3'),
-      new UnitTest('keypadPress number after number', keypadPress(3, true), '33'),
-
+    
+      new UnitTest('simulateFirstVisit()', simulateFirstVisit(true), '{}'),
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js:12]: In would pass in functions to be executed rather than passing results of tests into UnitTest. Then you can try catch in your unittest, show the failure and all the other tests can still execute

Comment (ultimatetag) [reply diceRoll/diceRollUnitTests.js:12]: (I would)

Comment (ultimatetag) [reply diceRoll/diceRollUnitTests.js:12]: new UnitTest('test name', function() { return myCode(); }, 'expected result');

function UnitTest(name, test, expected) {
    try { 
       var result = test();
.....

**Review by ultimatetag — APPROVED** [review]

File: diceRoll/diceRoll2.js (hunk 1 above)
Comment (EnderSmith) [reply diceRoll/diceRoll2.js:100]: Because I'm using it as a prototype? Is that incorrect form?

File: diceRoll/diceRoll2.js (hunk 2 above)
Comment (EnderSmith) [reply diceRoll/diceRoll2.js:109]: Okay. Thank you.

File: diceRoll/diceRoll2.js (hunk 3 above)
Comment (EnderSmith) [reply diceRoll/diceRoll2.js:153]: True/False (so that I remember it's a bool)

File: diceRoll/diceRollUnitTests.js (hunk 6 above)
Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:12]: Okay... I wasn't sure how to write unit tests, I'll see about doing that in a future request.
